### PR TITLE
[GraphBolt] add names for each item in ItemSet

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

added `Names` will be used to instantiate `MiniBatch`(previously called `DataBlock`).
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 1]
```diff
@@ -61,6 +61,53 @@ def test_OnDiskDataset_TVTSet_exceptions():
             _ = gb.OnDiskDataset(test_dir).load()
 
 
+def test_OnDiskDataset_TVTSet_ItemSet_names():
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:64]: Instead creating a new test case, the name attribute should be added for all other test cases.

Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:64]: I don't it's necessarily required as the name attributes parse/use logic are fully covered in the newly added testcase. No need to test it other existing testcases which will be redundant.

File: python/dgl/graphbolt/itemset.py [hunk 2]
```diff
@@ -47,11 +47,25 @@ class requires each input itemset to be iterable.
...
         else:
             self._items = (items,)
+        if names is not None:
+            if isinstance(names, tuple):
+                self._names = names
+            else:
+                self._names = (names,)
+            assert len(self._items) == len(
+                self._names
+            ), f"Number of items ({len(self._items)}) and names ({len(self._names)}) must match."
```
Comment (frozenbugs) [inline python/dgl/graphbolt/itemset.py:66]: line too long.

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py (hunk 1 above)
Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:64]: Test cases are also used by reader as an example of usage, it is better to align with the recommend way.

Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:64]: added.

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 3]
```diff
@@ -220,36 +326,45 @@ def test_OnDiskDataset_TVTSet_ItemSet_node_pair_label():
                 train_set:
                   - type: null
                     data:
-                      - format: numpy
+                      - name: src
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:329]: We need to make sure that user knows our keywords, currently the only mis-match between the reserved name and mini-batch variables is src, dst with node_pairs. Is this the best option?

**Review by frozenbugs — APPROVED** [review]

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py (hunk 3 above)
Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:329]: create a ticket for this issue: https://github.com/dmlc/dgl/issues/6259.

**Pull request merged**